core/src/main/java/com/linecorp/armeria/server/ServerBuilder.java

```diff
@@ -1952,6 +1955,18 @@ public ServerBuilder http1HeaderNaming(Http1HeaderNaming http1HeaderNaming) {
         return this;
     }
 
+    /**
+     * Sets the {@link AutoCloseable} which will be called whenever this {@link RequestContext} is popped
+     * from the {@link RequestContextStorage}.
+     *
+     * @param contextHook the {@link Supplier} that provides the {@link AutoCloseable}
+     */
+    public ServerBuilder contextHook(Supplier<? extends  AutoCloseable> contextHook) {
```

Comment:
Could you also add this method to `VirtualHostBuilder`?

Reply:
Our service and server builder become more complex due to `ContextPath*Builder`. A new skeleton builder may be necessary to check at compile-time and reduce duplication.

Reply:
Created for this https://github.com/line/armeria/issues/5236

Reply:
Can we use `ServiceConfigsBuilder` for this purpose?